### ISSUE TITLE
Не слышно голосовой чат игрока с Android

### ISSUE THREAD

**Issue by Vladislav4KZ** [FWGS/xash3d-fwgs#1111]
Запустил локальный сервер на ПК с Windows на последнем билде Xash3D FWGS и подключился к нему с телефона. И там я голосовой чат с телефона не слышу.

А когда подключился к серверу в Интернете, например на "Test Server Xash3D FWGS 0.20.1" где используется старый билд движка, голосовой чат с телефона работает.

**Comment by Vladislav4KZ** [comment]
На своём локальном сервере я не слышу звук с телефона вне зависимости от того, заглушил ли я себя с телефона или нет.

**Comment by Vladislav4KZ** [comment]
На телефоне у меня установлен Xash3D FWGS 0.21 с Play Маркета.

**Comment by a1batross** [comment]
Версия на Play Market не соответствует последнему мастеру отсюда. Замержим новый порт -- будет, но пока увы и ах.

Скорее всего это связано с попыткой исправить https://github.com/FWGS/xash3d-fwgs/issues/980

Что-то ему нужно ещё. Скорее всего game_shared часть из HLSDK, которая у нас была неосмотрительно вырезана.

**Comment by Vladislav4KZ** [comment]
Сейчас на Test Server Xash3D FWGS 0.20.1 стоит самый последний билд движка, такой же как и у меня на компьютере. Там я слышу себя когда включаю голосовой чат с телефона.

Но когда я создаю сервер в локальной сети со своего компьютера и подключаюсь к нему с телефона, я всё ещё не слышу голосовой чат с телефона на компьютере. Даже табличка с никнеймом не появляется.

**Comment by a1batross** [comment]
А hl.dll стимовский?

**Comment by Vladislav4KZ** [comment]
> А hl.dll стимовский?

Да, у меня на компьютере и на сервере "Test Server Xash3D FWGS 0.20.1"
@Elinsrc вчера говорил что он поставил оригинальную серверную библиотеку из Steam версии Half-Life на этот сервер.

Только у меня на компьютере стоит винда, а на сервере видимо линукс, только не уверен что дело в операционке. Раньше вроде и в локальной сети голосовой чат с телефона работал.

**Comment by Vladislav4KZ** [comment]
Теперь и на "Test Server Xash3D FWGS 0.20.1" перестал работать голосовой чат на Android.

**Comment by a1batross** [comment]
То он у вас работает, то не работает.

В sv_client_t есть поле 32-битное, по биту на каждого игрока. Надо смотреть в него. Скорее всего отладчиком на сервере.

Я по умолчанию его инициаилизирую во все единицы, чтобы HLSDK с вырезанным voice_gamemgr работал, хотя голдсорс сбрасывает в ноль.

voice_gamemgr с небольшим интервалом обновляет это поле через специальные функции, по принципу кто кого забанил, того и не слышит.

**Comment by Vladislav4KZ** [comment]
Перепроверил сейчас.

Когда подключен к удалённому серверу, (проверял на a1batross's server) голосовой чат игрока, играющего с Android версии движка – слышен.

На локальном сервере созданном на ПК с Windows, голосовой чат игрока, играющего на андроиде не слышно.

На локальном сервере, созданном на устройстве с Android, слышен голосовой чат игрока, играющего на андроиде.

Проверял на:
Xash3D FWGS v49/0.21
Номер сборки: 3418
Тип сборки: win32-i386
Коммит: https://github.com/FWGS/xash3d-fwgs/commit/6ee8a2dfa7060ded6f214b1b697e6dd4e8ab49ad
Ветка: master

ОС: Windows 10 Pro x64 22H2
Номер сборки ОС: 19045.4717

**Comment by Vladislav4KZ** [comment]
Если на ПК с Windows установить библиотеки собранные с репозитория [hlsdk-portable](https://github.com/FWGS/hlsdk-portable), и запустить локальный сервер на нём, то голосовой чат игрока с андроидом слышен.

Баг с не слышимым голосовым чатом игрока с Android, воспроизводится только на оригинальных игровых библиотеках Half-Life.

**Comment by SNMetamorph** [comment]
Всё ещё актуально?
